src/Text.tsx

```diff
@@ -1,13 +1,28 @@
+import React from 'react'
 import styled from 'styled-components'
 import {COMMON, SystemCommonProps, SystemTypographyProps, TYPOGRAPHY} from './constants'
 import sx, {SxProp} from './sx'
-import {ComponentProps} from './utils/types'
 
-const Text = styled.span<SystemTypographyProps & SystemCommonProps & SxProp>`
+const RestrictedAs = {
+  inlineText: ['span', 'p', 'b', 'i', 'em', 'small', 'strong', 'u', 'code', 'q', 's', 'label', 'div'],
+} as const
+
+const StyledText = styled.span<SystemTypographyProps & SystemCommonProps & SxProp>`
   ${TYPOGRAPHY};
   ${COMMON};
   ${sx};
 `
 
-export type TextProps = ComponentProps<typeof Text>
+export type TextProps = {
+  as?: typeof RestrictedAs.inlineText[number]
+} & React.ComponentProps<'span' & typeof StyledText>
+
+const Text = ({children, as = 'span', ...rest}: TextProps) => {
+  return (
+    <StyledText as={RestrictedAs.inlineText.includes(as) ? as : undefined} {...rest}>
```

Comment:
In development mode, could we also output an error in the console if the developer passed an invalid `as` value? If we're ignoring their value, we should warn them about it. @green6erry has code for this.

Reply:
Good call, I forgot that there might be confusion on why applying an element isn't working as expected.

Reply:
Should now be added. I added a message around what elements are allowed, though I want to make sure it isn't too verbose.